Jenkinsfile

```diff
@@ -1,42 +1,20 @@
-@Library(['ableton-utils@0.1.0', 'groovylint@0.1.1']) _
+@SuppressWarnings('VariableTypeRequired') // For the declaration of the _ variable
+@Library(['ableton-utils@0.3.0', 'groovylint@0.3.0']) _
 
 
-def addStages() {
-  runTheBuilds.timedStage('Checkout') {
-    // Print out all environment variables for debugging purposes
-    sh 'env'
-    checkout scm
-  }
-
-  runTheBuilds.timedStage('Test') {
+runTheBuilds.runDevToolsProject(script: this,
+  test: {
     parallel(failFast: false,
       groovylint: {
         groovylint.check('./Jenkinsfile')
       },
       junit: {
-        sh './gradlew test'
-        junit 'build/test-results/**/*.xml'
+        try {
+          sh './gradlew test'
+        } finally {
```

Comment:
why is this in a finally block?

Reply:
☝️ by default, `sh` throws when what it runs returns non-zero.

Reply:
Yeah, what @ala-ableton said. Basically for failed tests the build would have the correct result (failed), but you wouldn't see the failing unit test in the blue ocean view.

Reply:
#TIL that this could also be expressed like so:

```groovy
catchError {
  sh './gradlew test'
}
junit 'build/**/*.xml'
```

`catchError` is a Jenkins pipeline function, so I kinda like `try/finally` better since it's more universally understood. However, just saying, we could also do it this way.